src/main/java/seedu/duke/model/lesson/LessonList.java

```diff
@@ -38,7 +39,13 @@ public boolean isEmpty() {
         return lessonList.isEmpty();
     }
 
-    public void addLesson(Lesson newLesson) {
+    public void addLesson(Lesson newLesson) throws DukeException {
+        for (Lesson lesson : lessonList) {
+            if (lesson.getTitle().equals(newLesson.getTitle())
+                    && (lesson.getDayOfTheWeek().equals(newLesson.getDayOfTheWeek()))) {

```

Comment:
Should probably check if the start and end time are similar as well? Also, consider refactoring the logic inside the `if` for SLAP.